packages/core/src/plugin/index.ts

```diff
@@ -9,8 +9,27 @@ import merge from '../utils/merge.js';
 export * from './js/index.js';
 export * from './rust/index.js';
 
-export async function resolveFarmPlugins(config: UserConfig) {
-  const plugins = config.plugins ?? [];
+export async function resolveAsyncPlugins<T>(arr: T[]): Promise<T[]> {
```

Comment:
This method seems to just move the position, and we should put the important method at the top of the code